src/index.js

```diff
@@ -154,19 +155,41 @@ function renderPdf(tag) {
   }
 
   const result = {};
+  const isTopLevel = true;
 
   children.forEach(child => {
     const resolvedChild = resolve(child, context);
     result[resolvedChild.elementName] = resolveChildren(
       resolvedChild,
       context,
-      true,
+      isTopLevel,
     );
   });
+  let { header, footer } = result;
+
+  if (has(attributes, 'header') && typeof attributes.header === 'function') {
+    header = (currentPage, pageCount, pageSize) =>
+      resolveChildren(
+        attributes.header(currentPage, pageCount, pageSize),
+        context,
+        isTopLevel,
+      );
+  }
+
+  if (has(attributes, 'footer') && typeof attributes.footer === 'function') {
```

Comment:
Might be an opportunity for some deduplication of the header and footer code here.